### PR TITLE
split DB interface into IdentityStore and TokenStore

### PR DESCRIPTION
**Pull request by daiLlew**
Split DB interface into 2 separate interfaces `IdentityStore` and `TokenStore`.

### CODE REVIEW

File: persistence/token_db_wrapper.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+	"time"
+)
+
+var nilID = ""
+
+type Cache interface {
+	StoreToken(key string, i schema.Identity, ttl time.Duration) error
+	GetToken(token string) (time.Duration, error)
+}
+type CacheWrapper struct {
```
Comment (daiLlew) [inline persistence/token_db_wrapper.go]: I think we could rename this to something like `CachedTokenStored` now as its specifically a cache layer around the token store.

File: persistence/token_db_wrapper.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+	TokenCache Cache
```
Comment (daiLlew) [inline persistence/token_db_wrapper.go]: Could just call it `Cache`.

File: persistence/token_db_wrapper.go [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+	TokenDb    TokenStore
```
Comment (daiLlew) [inline persistence/token_db_wrapper.go]: Following go naming conventions it should be `TokenDB`

**Review by daiLlew — COMMENTED** [review]
APPROVED.

**Review by mikeAdamss — APPROVED** [review]

**Pull request merged**